doc/sqa/sdd/ravenSoftwareDesing.tex

```diff
@@ -0,0 +1,478 @@
+\section{Software Design}
+\subsection{Introduction}
+The Risk Analysis and Virtual ENviroment (RAVEN) mission is to provide a framework/container of capabilities for 
+engineers and scientists to analyze the response of systems, physics and multi-physics, employing advanced numerical
+techniques and algorithms.
+RAVEN was conceived with two major objectives: 
+\begin{itemize}
+  \item  to be as easy and straightforward to use by scientists and engineers as possible. 
+  \item to allow its expansion in a straightforward manner, providing clear and modular APIs  to developers.
+\end{itemize}
+
+The RAVEN software is meant to be approachable by any type of user (computational scientists, engineers and analysts). 
+Every single aspect of RAVEN was driven by this singular principle from the build system to the APIs to the software development cycle and input syntax.
+
+\begin{figure}[ht]
+  \centering
+  \includegraphics[width=0.9\textwidth]  {pics/RavenFramework.png}
+  \caption{RAVEN framework layout}
+  \label{fig:RAVENframeworkLayout}
+\end{figure}
+
+The main idea behind the design of the RAVEN software was/is the creation of a multi-purpose framework characterized by 
+high flexibility with respect to the possible perform-able analysis. The framework must be capable of constructing the 
+analysis/calculation flow at run-time, interpreting the user-defined instructions and assembling the different analysis tasks 
+following a user specified scheme.
+In order to achieve such flexibility, combined with reasonably fast development, a programming language naturally suitable 
+for this kind of approach was needed: $Python$.
+Hence, RAVEN is coded in $Python$ and is characterized by an object-oriented design. 
```

Comment:
This does not need to be fixed, but $ should generally only be used for equations, \emph or \textit would be better for italics.

Reply:
fixed